### PR TITLE
Arizona Military Pension Subtraction

### LINKED ISSUE #3004: Arizona Benefits, Annuities and Pensions Subtraction
https://www.azleg.gov/viewdocument/?docName=https://www.azleg.gov/ars/43/01022.htm

### PR DESCRIPTION
**Pull request by JudyZ13**
Fixes #3004

### Summary
:sparkles::card_file_box::pencil:

This pull request adds two new features to the Arizona income tax system: a subtraction for benefits, annuities and pensions, and a subtraction for long-term net capital gains. It creates new parameter, variable and test files for each feature, and updates the changelog accordingly.

> _New subtractions add_
> _`Arizona` tax logic_
> _Autumn leaves fall fast_

### Walkthrough
*  Add Arizona benefits, annuities and pensions subtraction as a new feature ([link](https://github.com/PolicyEngine/policyengine-us/pull/3005/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
   * Create a new parameter file `amount.yaml` to define the subtraction threshold by year and filing status ([link](https://github.com/PolicyEngine/policyengine-us/pull/3005/files?diff=unified&w=0#diff-17e91a60a452010d157a08e8b3443c09d01c128a04cf5dbd4b4532fac201980eR1-R16))
   * Create a new variable file `az_benefits_annuities_pensions_subtraction.py` to implement the subtraction logic based on the parameter values and the tax unit's income sources ([link](https://github.com/PolicyEngine/policyengine-us/pull/3005/files?diff=unified&w=0#diff-f21515066961f2258cf95aae63ed5b1eadafa1f8cc17bd7d531b826f1371153dR1-R30))
   * Create a new test file `az_benefits_annuities_pensions_subtraction.yaml` to verify the subtraction calculation for different scenarios ([link](https://github.com/PolicyEngine/policyengine-us/pull/3005/files?diff=unified&w=0#diff-ccc4fe1666f731b0a38fc8ee2fd731ba90185bb9dee9c941f774c4e4c6e788d7R1-R49))
* Add Arizona net capital gains subtraction as a new feature ([link](https://github.com/PolicyEngine/policyengine-us/pull/3005/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
   * Create a new parameter file `rate.yaml` to define the subtraction rate by year ([link](https://github.com/PolicyEngine/policyengine-us/pull/3005/files?diff=unified&w=0#diff-e3a63ce81ad3d884795a17f131e4769a3b8972122c535979ad90128b36d7fbacR1-R16))
   * Create a new variable file `az_net_capital_gains_subtraction.py` to implement the subtraction logic based on the parameter values and the tax unit's capital gains ([link](https://github.com/PolicyEngine/policyengine-us/pull/3005/files?diff=unified&w=0#diff-7101277d60ac269a1c7e5fb64fa1d6b60efc09ebdf4b2df0bf5dfe3128ea1832R1-R22))
   * Create a new test file `az_net_capital_gains_subtraction.yaml` to verify the subtraction calculation for different scenarios ([link](https://github.com/PolicyEngine/policyengine-us/pull/3005/files?diff=unified&w=0#diff-b3f269c10e1e00ab1ca0cb7a20b15975056cb52f8c1e302c39293fa450b925e9R1-R15))

### CODE REVIEW

File: policyengine_us/variables/gov/states/az/tax/income/deductions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from policyengine_us.model_api import *
+
+
+class az_benefits_annuities_pensions_subtraction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Arizona benefits, annuitites and pensions subtraction"
+    unit = USD
+    documentation = ""
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:9]:
```suggestion
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/benefits_annuities_pensions/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  unit: currency-USD
+  reference:
+  - title: 43-1022. Subtractions from Arizona gross income (26)
+    href: https://www.azleg.gov/ars/43/01022.htm
+  - title: 2022 Arizona Resident Personal Income Tax Return Form140 Line29(b)
+    href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140.pdf#page=1
+  - title: 2022 Arizona Resident Personal Income Tax Form140 Instructions
+    href: https://azdor.gov/sites/default/files/2023-03/FORMS_INDIVIDUAL_2022_140i.pdf#page=15
+values:
+  0000-01-01: 2_500
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/benefits_annuities_pensions/amount.yaml:14]:
```suggestion
  2018-01-01: 2_500
```

File: policyengine_us/variables/gov/states/az/tax/income/deductions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        ).gov.states.az.tax.income.deductions.benefits_annuities_pensions
+        person = tax_unit.members
+
+        filing_status = tax_unit("filing_status", period)
+        married = filing_status == filing_status.possible_values.JOINT
+        subtraction_amount = min_(
+            p.amount, person("military_retirement_pay", period)
+        )
+
+        subtraction_joint = tax_unit.sum(subtraction_amount)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:25]: just head and spouse - not other dependents, need to add that condition

File: policyengine_us/variables/gov/states/az/tax/income/deductions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.az.tax.income.deductions.benefits_annuities_pensions
+        person = tax_unit.members
+
+        filing_status = tax_unit("filing_status", period)
+        married = filing_status == filing_status.possible_values.JOINT
+        subtraction_amount = min_(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/deductions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:21]:
```suggestion
        military_retirement_pay = min_(
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/benefits_annuities_pensions/amount.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+description: Arizona excludes benefits, annuities and pensions if the amount is under the following threshold.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/benefits_annuities_pensions/amount.yaml:1]:
```suggestion
description: Arizona excludes benefits, annuities and pensions from uniformed services of the United States of up to this amount.
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/benefits_annuities_pensions/amount.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+metadata:
+  label: Arizona benefits, annuities and pensions subtraction threshold
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/benefits_annuities_pensions/amount.yaml:3]:
```suggestion
  label: Arizona benefits, annuities and pensions subtraction max amount
```

File: policyengine_us/parameters/gov/states/az/tax/income/deductions/benefits_annuities_pensions/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/deductions/benefits_annuities_pensions/amount.yaml:1]: rename to max_amount.yaml

File: changelog_entry.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Arizona benefits, annuities and pensions subtraction.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Arizona military pension subtraction.
```

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/max_amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/max_amount.yaml:1]: rename to cap

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/max_amount.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+description: Arizona excludes benefits, annuities and pensions from uniformed services of the United States of up to this amount.
+metadata:
+  label: Arizona benefits, annuities and pensions subtraction max amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/max_amount.yaml:3]:
```suggestion
  label: Arizona benefits, annuities and pensions subtraction cap
```

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
+from policyengine_us.model_api import *
+
+
+class az_benefits_annuities_pensions_subtraction(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:4]:
```suggestion
class az_military_pension_subtraction(Variable):
```

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Arizona benefits, annuitites and pensions subtraction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:7]:
```suggestion
    label = "Arizona military pension subtraction"
```

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py (hunk 9 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:4]: rename .py and test file accordingly

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+            p.max_amount, person("military_retirement_pay", period)
+        )
+        head = person("is_tax_unit_head", period)
+        spouse = person("is_tax_unit_spouse", period)
+
+        subtraction_joint = tax_unit.sum(
+            military_retirement_pay * (head + spouse)
+        )
+
+        subtraction_other = tax_unit.sum(military_retirement_pay * head)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:31]:
```suggestion
```

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+        filing_status = tax_unit("filing_status", period)
+        married = filing_status == filing_status.possible_values.JOINT
+        military_retirement_pay = min_(
+            p.max_amount, person("military_retirement_pay", period)
+        )
+        head = person("is_tax_unit_head", period)
+        spouse = person("is_tax_unit_spouse", period)
+
+        subtraction_joint = tax_unit.sum(
+            military_retirement_pay * (head + spouse)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:28]:
```suggestion
            military_retirement_pay * (head | spouse)
```

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+        filing_status = tax_unit("filing_status", period)
+        married = filing_status == filing_status.possible_values.JOINT
+        military_retirement_pay = min_(
+            p.max_amount, person("military_retirement_pay", period)
+        )
+        head = person("is_tax_unit_head", period)
+        spouse = person("is_tax_unit_spouse", period)
+
+        subtraction_joint = tax_unit.sum(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:27]:
```suggestion
        return tax_unit.sum(
```

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+            military_retirement_pay * (head + spouse)
+        )
+
+        subtraction_other = tax_unit.sum(military_retirement_pay * head)
+
+        return where(married, subtraction_joint, subtraction_other)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:33]:
```suggestion
```

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+    defined_for = StateCode.AZ
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.az.tax.income.subtractions.benefits_annuities_pensions
+        person = tax_unit.members
+
+        filing_status = tax_unit("filing_status", period)
+        married = filing_status == filing_status.possible_values.JOINT
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/az/tax/income/subtractions/benefits_annuities_pensions/az_benefits_annuities_pensions_subtraction.py:20]:
```suggestion
```

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/military_pensions/az_military_pension_subtraction.py [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+        p = parameters(
+            period
+        ).gov.states.az.tax.income.subtractions.military_pension
+        person = tax_unit.members
+
+        filing_status = tax_unit("filing_status", period)
+        military_retirement_pay = min_(
+            p.cap, person("military_retirement_pay", period)
+        )
+        head = person("is_tax_unit_head", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/az/tax/income/subtractions/military_pensions/az_military_pension_subtraction.py:23]:
```suggestion
        head_or_spouse = person("is_tax_unit_head_or_spouse", period)
```

File: policyengine_us/variables/gov/states/az/tax/income/subtractions/military_pensions/az_military_pension_subtraction.py [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+    definition_period = YEAR
+    defined_for = StateCode.AZ
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.az.tax.income.subtractions.military_pension
+        person = tax_unit.members
+
+        filing_status = tax_unit("filing_status", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/az/tax/income/subtractions/military_pensions/az_military_pension_subtraction.py:19]: remove - unnecessary

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/military_pension/cap.yaml [hunk 18]
```diff
@@ -0,0 +1,16 @@
+description: Arizona excludes benefits, annuities and pensions from uniformed services of the United States of up to this amount.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/military_pension/cap.yaml:1]: To clarify that it's not all military benefits, or all pensions among military retirees.
```suggestion
description: Arizona subtracts from gross income any benefits, annuities and pensions as retired or retainer pay of the uniformed services of the United States, up to this amount.
```

File: policyengine_us/parameters/gov/states/az/tax/income/subtractions/military_pension/cap.yaml [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+metadata:
+  label: Arizona military pensions subtraction max amount
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/az/tax/income/subtractions/military_pension/cap.yaml:3]:
```suggestion
  label: Arizona military pensions subtraction cap
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Comment by PavelMakarchuk** [comment]
Duplicate

**Pull request closed without merging**